File: Backend/Domain/Motherboard.cs
using System;

namespace Domain
{
    public class Motherboard
    {
        public int Id { get; set; }
        public Part Part { get; set; }
        public Socket Socket { get; set; }
        public FormFactor FormFactor { get; set; }
        public RamType RamType { get; set; }
        public string Chipset { get; set; }
        public bool Oc { get; set; }
        public bool Rgb { get; set; }
    }
}

File: Frontend/src/app/models/Motherboard.ts
import {Part} from "./Part";
import {FormFactor} from "./FormFactor";
import {Socket} from "./Socket";
import {RamType} from "./RamType";

export interface Motherboard {
    id: number;
    part: Part;
    socket: Socket;
    formFactor: FormFactor
    ramType: RamType;
    chipset: string;
    oc: boolean;
    rgb: boolean;
}
